pkg/motor/bucket_resolver.go

```diff
@@ -0,0 +1,118 @@
+package motor
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sonr-io/sonr/internal/bucket"
+	"github.com/sonr-io/sonr/pkg/motor/types"
+	bt "github.com/sonr-io/sonr/x/bucket/types"
+)
+
+func (mtr *motorNodeImpl) NewBucketResolver(context context.Context, did string) (bucket.Bucket, error) {
+	if did == "" {
+		return nil, errors.New("creator address and did must be defined within the request")
+	}
+	addr, err := mtr.Wallet.Address()
+	if err != nil {
+		return nil, err
+	}
+
+	if _, ok := mtr.Resources.whereIsStore[did]; !ok {
+		wiReq, err := mtr.QueryWhereIs(context, types.QueryWhereIsRequest{
+			Creator: addr,
+			Did:     did,
+		})
+
+		if err != nil {
+			return nil, err
+		}
+
+		mtr.Resources.whereIsStore[did] = wiReq.WhereIs
+	}
+	wi := mtr.Resources.whereIsStore[did]
+	s := mtr.Resources.shell
+	bq := mtr.Resources.bucketQueryClient
+
+	b := bucket.New(addr, wi, s, bq)
```

Comment:
nit: making local variables for each parameter here makes this less readable. Consider either making the variable names more obvious or just putting the parameters inline